common/swagger/v2/swagger.yaml

```diff
@@ -0,0 +1,71 @@
+openapi: '3.0.0'
+info:
+  description: >-
+    This is an extension of the REST API for scoring from H2O. This API is
+    intended to be used for scoring within H2O.ai MLOps.
+  version: 2.0.0
+  title: Scoring API - v2
+  contact:
+    email: support@h2o.ai
+  license:
+    name: License
+    url: 'http://www.h2o.ai'
+servers:
+  - url: '{protocol}://127.0.0.1:{port}/'
+    variables:
+      protocol:
+        enum:
+          - http
+          - https
+        default: http
+      port:
+        default: '8080'
+paths:
+  /model/score/media:
+    post:
+      tags:
+        - scoring
+      summary: Score model with provided media files
+      description: Computes score of provided data making use of provided media files
+      operationId: getScoreMedia
+      requestBody:
+        content:
+          multipart/form-data:
+            schema:
+              type: object
+              properties:
+                payload:
+                  $ref: '#/components/schemas/ScoreMediaRequest'
+                files:
```

Comment:
Does this mean that there must be a field labeled `payload` in the POST call?  What is the type of that field?  Is it an object itself, or a JSON string?

Would it make sense to flatten the structure so that those fields are on the same level as `files`?

Reply:
yes the request would take the format:
```python
res = requests.post("http://localhost:8080/model/score/media", data={"payload": json.dumps(req)}, files=files)
```
using the requests package in python.

Sadly, no this is not possible. Form data does not support complex objects like `list of list` therefore it is not possible to define the payload as independent form fields.

Reply:
@orendain

Reply:
ummmm... @orendain just to be clear `shapley contributions` != `image/media scoring`

sure we could flatten the payload, I guess the question becomes, do we want to validate a single string, or multiple different strings, in the backend, you'd end up having to parse 2 strings using some version of `json.loads()` to get fields which is an array of strings, and rows which is an array of arrays of strings. Personally, I don't like this... like from client call perspective, it increases the amount of work required to construct the proper payload request, and does not (imho) significantly improve the ability to understand or utilize the api.